### PR TITLE
Update iota identity

### PR DESCRIPTION
**Pull request by 7opf**
* Updated IOTA Identity WASM to 0.4.0
* Fixed breaking changes
* Updated tutorial docs

### CODE REVIEW

File: identity-enabler/deviceId-mobile-app/.gitignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/public/build/
+/public/*.wasm
+.gradle
+/android/.idea
+
+# Capacitor config
+android/app/src/main/assets/capacitor.config.json
```
Comment (jmcanterafonseca-iota) [inline identity-enabler/deviceId-mobile-app/.gitignore:7]: new line needed

File: identity-enabler/deviceId-mobile-app/src/services/identityService.ts [hunk 2]
```diff
@@ -2,285 +2,311 @@ import Keychain from '../lib/keychain';
...
-            publicAuthKey: key.public,
-            privateAuthKey: key.secret,
-            doc,
-            key,
-            keys,
-            method,
-        };
+  private client?: identity.Client;
+
+  constructor(private readonly devnet: boolean, private readonly nodeUrl: string) {}
```
Comment (jmcanterafonseca-iota) [inline identity-enabler/deviceId-mobile-app/src/services/identityService.ts:25]: there are better design options here for modelling the usage of a network rather than using a boolean

Comment (jmcanterafonseca-iota) [reply identity-enabler/deviceId-mobile-app/src/services/identityService.ts:25]: for instance just passing a string with the name of the network as it was before ...

File: identity-enabler/deviceId-mobile-app/src/services/identityService.ts [hunk 3]
```diff
@@ -2,285 +2,311 @@ import Keychain from '../lib/keychain';
...
+  ): Promise<identity.VerifiableCredential> {
+    // Initialize the Library - Is cached after first initialization
+    await identity.init();
+
+    // Prepare credential Data
+    const IssuerDidDoc = Document.fromJSON(JSON.parse(issuer.didDoc));
+    const IssuerKeys = KeyCollection.fromJSON(issuer.keys);
+    const IssuerDoc = Document.fromJSON(issuer.doc);
+    const IssuerMethod = VerificationMethod.fromJSON(issuer.method);
+
```
Comment (jmcanterafonseca-iota) [inline identity-enabler/deviceId-mobile-app/src/services/identityService.ts:143]: name of variables shall start with lower case

File: Identity.md [hunk 4]
```diff
@@ -5,15 +5,10 @@
...
-import type { IdentityConfig } from '../models/types/identity';    
 
-const config: IdentityConfig = {
-	  node: IOTA_NODE_URL,
-	  network: DEVNET ? 'dev' : 'main',
-};
 
 export default () => {
-      ServiceFactory.register('identity', () => new IdentityService(config));  
+      ServiceFactory.register('identity', () => new IdentityService(DEVNET, IOTA_NODE_URL));
```
Comment (jmcanterafonseca-iota) [inline Identity.md:11]: is this change really needed?

File: identity-enabler/deviceId-mobile-app/src/lib/init.ts [hunk 5]
```diff
@@ -1,13 +1,7 @@
...
-const config: IdentityConfig = {
-	node: IOTA_NODE_URL,
-	network: DEVNET ? 'dev' : 'main',
-};
 
 export default () => {
-    ServiceFactory.register('identity', () => new IdentityService(config));  
-}
+  ServiceFactory.register('identity', () => new IdentityService(DEVNET, IOTA_NODE_URL));
+};
```
Comment (jmcanterafonseca-iota) [inline identity-enabler/deviceId-mobile-app/src/lib/init.ts:7]: same comment

File: identity-enabler/deviceId-mobile-app/src/models/types/identity.ts [hunk 6]
```diff
@@ -1,18 +1,13 @@
...
-  method: typeof IotaIdentity.VerificationMethod | any;
-};
-
-export type IdentityConfig = {
-  node: string;
-  network: string;
+  doc: Document;
+  key: KeyPair;
+  keys: KeyCollection | any;
+  method: VerificationMethod | any;
```
Comment (jmcanterafonseca-iota) [inline identity-enabler/deviceId-mobile-app/src/models/types/identity.ts:10]: can we avoid anys?

File: identity-enabler/deviceId-mobile-app/src/services/identityService.ts [hunk 7]
```diff
@@ -2,285 +2,311 @@ import Keychain from '../lib/keychain';
...
+    // Add to the DID Document as a general-purpose verification method
+    doc.insertMethod(method, 'VerificationMethod');
+
+    // Signing
+    doc.sign(key);
+
+    // Publish
+    await this.getClient().publishDocument(doc);
+    return {
+      didDoc: JSON.stringify(doc.toJSON()),
```
Comment (jmcanterafonseca-iota) [inline identity-enabler/deviceId-mobile-app/src/services/identityService.ts:71]: is it really needed to stringify

**Comment by 7opf** [comment]
Closing in favour of https://github.com/jmcanterafonseca-iota/zebra-iota-edge-sdk/pull/28/files
Separate PR for refactor (could remove capacitor simultaneously?)
